### PR TITLE
151 javadocs

### PR DESCRIPTION
**Pull request by smolamarcin**

### CODE REVIEW

File: battleship-server/src/main/java/com/epam/solid/nie/config/ConfigProperty.java [hunk 1]
```diff
@@ -1,5 +1,10 @@
 package com.epam.solid.nie.config;
 
+/**
+ * Variables corresponding to variables in the game.
+ * They specify the server parameters to which we connect, language in the game
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/epam/solid/nie/config/ConfigProperty.java]: maybe ',' at the end?

File: battleship-server/src/main/java/com/epam/solid/nie/config/ConfigProperty.java [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+ * output type and number of rounds in the game.
+ */
 public enum ConfigProperty {
     SERVER_PORT, SERVER_IP, BOARD_SIZE, ROUNDS, FOUR_MAST_SHIP, THREE_MAST_SHIP, TWO_MAST_SHIP, ONE_MAST_SHIP, OUTPUT, LANGUAGE
 }
```
Comment (rivalez) [inline battleship-server/src/main/java/com/epam/solid/nie/config/ConfigProperty.java]: new line

File: battleship-server/src/main/java/com/epam/solid/nie/config/ConfigProperty.java [hunk 3]
```diff
@@ -1,5 +1,10 @@
 package com.epam.solid.nie.config;
 
+/**
+ * Variables corresponding to variables in the game.
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/epam/solid/nie/config/ConfigProperty.java]: Can you change one of this 'variables'?(Like first one to enums maybe)

File: battleship-server/src/main/java/com/epam/solid/nie/config/FileConfiguration.java [hunk 4]
```diff
@@ -6,10 +6,18 @@
...
+ * Reads configuration from the file.
+ */
 public class FileConfiguration implements Configuration {
 
     private EnumMap<ConfigProperty, String> map = new EnumMap(ConfigProperty.class);
 
+    /**
+     * Return the config map which is used in the game.
+     * The map is filled based on the configuration file.
+     * @return
```
Comment (rivalez) [inline battleship-server/src/main/java/com/epam/solid/nie/config/FileConfiguration.java]: describe or delete empty @return

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/Communicate.java [hunk 5]
```diff
@@ -1,5 +1,8 @@
 package com.epam.solid.nie.config.language;
 
+/**
+ * Each constant represents the communicate displayed in the game.
+ */
 public enum Communicate {
     WELCOME
 }
```
Comment (rivalez) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/Communicate.java]: new line

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProvider.java [hunk 6]
```diff
@@ -1,5 +1,8 @@
 package com.epam.solid.nie.config.language;
 
+/**
+ * It provides the correct communicates based on specified language.
+ */
 public interface CommunicateProvider {
     CommunicateProviderImpl populate(Language language);
 }
```
Comment (rivalez) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProvider.java]: new line

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProviderImpl.java [hunk 7]
```diff
@@ -6,9 +6,17 @@
...
+/**
+ * It provides relevant communicates to the game.
+ */
 public class CommunicateProviderImpl implements CommunicateProvider {
     private Map<Communicate, String> communicates = new HashMap<>();
 
+    /**
+     * Fills the map with messages in the appropriate language.
+     * @param language
+     * @return
```
Comment (rivalez) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProviderImpl.java]: describe/delete @return the same with language

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/Language.java [hunk 8]
```diff
@@ -1,5 +1,8 @@
 package com.epam.solid.nie.config.language;
 
+/**
+ * Available languages in the game.
+ */
 public enum Language {
     POLISH, ENGLISH
 }
```
Comment (rivalez) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/Language.java]: new line

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/LanguageConfiguration.java [hunk 9]
```diff
@@ -1,5 +1,9 @@
 package com.epam.solid.nie.config.language;
 
-public interface LanguageConfiguration {
+/**
+ * Implementation should provide actions responsible
+ * for provide specific language to the game.
+ */
+interface LanguageConfiguration {
     Language provide();
 }
```
Comment (rivalez) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/LanguageConfiguration.java]: new line

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/LanguageConfigurationImpl.java [hunk 10]
```diff
@@ -4,10 +4,14 @@
...
 import java.util.ResourceBundle;
 
-public class LanguageConfigurationImpl implements LanguageConfiguration {
+class LanguageConfigurationImpl implements LanguageConfiguration {
 
     private static final String LANGUAGE = "LANGUAGE";
 
+    /**
+     * Returns the appropriate language that is loaded from the configuration file.
+     * @return
```
Comment (rivalez) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/LanguageConfigurationImpl.java]: describe/delete @return

File: common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java [hunk 11]
```diff
@@ -15,19 +15,33 @@
...
 
-    HorizontalShip addType(){
+    HorizontalShip addType() {
         this.battleShipType = BattleShipType.valueOf(getShipsRemainingHealth(getPositions()));
         return this;
     }
 
+    /**
+     * Returns what type of ship is there (number of masts)
+     * @return
```
Comment (rivalez) [inline common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java]: describe/delete @return

File: common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java [hunk 12]
```diff
@@ -15,19 +15,33 @@
...
+     */
     public BattleShipType getBattleShipType() {
         return battleShipType;
     }
 
+    /**
+     * Returns the list of fields (represents by points, which contains coordinates)
+     * that are occupied by the ship.
+     * @return
```
Comment (rivalez) [inline common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java]: describe/delete @return

File: common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java [hunk 13]
```diff
@@ -15,19 +15,33 @@
...
+     */
     public List<Point2D> getShipProperties() {
         return getPositions();
     }
 
+    /**
+     * Returns the number of masts that have not yet been shot based on a fields occupied by the ship.
+     * @param positions - list of fields occupied by the ship
+     * @return
```
Comment (rivalez) [inline common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java]: describe/delete @return

File: common/src/main/java/com/epam/solid/nie/ships/HorizontalShipFactory.java [hunk 14]
```diff
@@ -8,6 +8,11 @@
  * Created by marek on 13.12.2017.
  */
 public class HorizontalShipFactory implements ShipFactory {
+    /**
+     * Create an instance of horizontal ship based on a list of points
+     * @param positions
+     * @return
```
Comment (rivalez) [inline common/src/main/java/com/epam/solid/nie/ships/HorizontalShipFactory.java]: describe/delete @return

File: common/src/main/java/com/epam/solid/nie/ships/VerticalShip.java [hunk 15]
```diff
@@ -15,15 +15,26 @@
...
-    VerticalShip addType(){
+    VerticalShip addType() {
         this.battleShipType = BattleShipType.valueOf(getShipsRemainingHealth(getPositions()));
         return this;
     }
 
+    /**
+     * Returns what type of ship is there (number of masts)
+     *
+     * @return
```
Comment (rivalez) [inline common/src/main/java/com/epam/solid/nie/ships/VerticalShip.java]: describe/delete @return

File: common/src/main/java/com/epam/solid/nie/ships/VerticalShipFactory.java [hunk 16]
```diff
@@ -8,6 +8,11 @@
  * Created by marek on 13.12.2017.
  */
 public class VerticalShipFactory implements ShipFactory {
+    /**
+     * Create an instance of vertical ship based on a list of points
+     * @param positions
+     * @return
```
Comment (rivalez) [inline common/src/main/java/com/epam/solid/nie/ships/VerticalShipFactory.java]: describe/delete @return

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/Communicate.java [hunk 17]
```diff
@@ -1,5 +1,8 @@
 package com.epam.solid.nie.config.language;
 
+/**
+ * Each constant represents the communicate displayed in the game.
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/Communicate.java]: Isn't 'communicate' verb? Maybe you meant 'information', or 'communication'

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProvider.java [hunk 18]
```diff
@@ -1,5 +1,8 @@
 package com.epam.solid.nie.config.language;
 
+/**
+ * It provides the correct communicates based on specified language.
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProvider.java]: Maybe 'proper' instead of 'correct'?

Comment (Ozymadias) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProvider.java]: 'communicates' as above

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProviderImpl.java [hunk 19]
```diff
@@ -6,9 +6,17 @@
 import java.util.function.Consumer;
 import java.util.stream.Stream;
 
+/**
+ * It provides relevant communicates to the game.
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProviderImpl.java]: 'communicates' as above

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProviderImpl.java [hunk 20]
```diff
@@ -20,10 +28,19 @@ public CommunicateProviderImpl populate(Language language) {
                 .name()));
     }
 
+    /**
+     * Retrieves the message in the appropriate language from the map.
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProviderImpl.java]: Maybe it isn't necessary to tell about structure used in implementation

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProviderImpl.java [hunk 21]
```diff
@@ -20,10 +28,19 @@ public CommunicateProviderImpl populate(Language language) {
...
+     * @param communicate
+     * @return specified communicate from the map
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/CommunicateProviderImpl.java]: again 'communicate'

File: battleship-server/src/main/java/com/epam/solid/nie/config/language/LanguageConfiguration.java [hunk 22]
```diff
@@ -1,5 +1,9 @@
 package com.epam.solid.nie.config.language;
 
-public interface LanguageConfiguration {
+/**
+ * Implementation should provide actions responsible
```
Comment (Ozymadias) [inline battleship-server/src/main/java/com/epam/solid/nie/config/language/LanguageConfiguration.java]: Maybe shorter like: "Provides games language"? Now you use 'provide two times.

File: common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java [hunk 23]
```diff
@@ -15,19 +15,33 @@
...
+     * that are occupied by the ship.
+     * @return
+     */
     public List<Point2D> getShipProperties() {
         return getPositions();
     }
 
+    /**
+     * Returns the number of masts that have not yet been shot based on a fields occupied by the ship.
+     * @param positions - list of fields occupied by the ship
```
Comment (Ozymadias) [inline common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java]: Shouldn't it be: 'have not been shot yet'

File: common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java [hunk 24]
```diff
@@ -15,19 +15,33 @@
...
+     * Returns the list of fields (represents by points, which contains coordinates)
+     * that are occupied by the ship.
+     * @return
+     */
     public List<Point2D> getShipProperties() {
         return getPositions();
     }
 
+    /**
+     * Returns the number of masts that have not yet been shot based on a fields occupied by the ship.
```
Comment (Ozymadias) [inline common/src/main/java/com/epam/solid/nie/ships/HorizontalShip.java]: Is 'based on ...' part really necessary?

File: common/src/main/java/com/epam/solid/nie/state/PlayState.java [hunk 25]
```diff
@@ -5,7 +5,10 @@
 
 @ToString
 public class PlayState implements State {
-
+    /**
+     * Changes the current state.
```
Comment (Ozymadias) [inline common/src/main/java/com/epam/solid/nie/state/PlayState.java]: Many same information of type: 'Changes the current state.' :/

**Pull request merged**